src/behaviours/utils/block-viewport-scroll.ts

```diff
@@ -1,16 +1,17 @@
-import { safeWindow } from "../../events/utils/window"
-
 let isViewportScrollBlocked = false
+const isBrowser = typeof window !== "undefined"
 
-safeWindow.addEventListener(
-    "touchmove",
-    (event: TouchEvent) => {
-        if (isViewportScrollBlocked) {
-            event.preventDefault()
-        }
-    },
-    { passive: false }
-)
+if (isBrowser) {
+    window.addEventListener(
+        "touchmove",
```

Comment:
Yeah I believe there was a reason for this - something like if you add the event listener on `touchstart`, you aren't then able to block subsequent `touchmove` events. But if you always had the listener on and toggled this boolean it would work.

I'd like to revisit this with a PR for conditionally blocking the viewport scroll based on scroll direction but that is a project in itself.

Reply:
I think we should figure out what the exact reason was and add it as a comment

Reply:
I agree but that investigation doesn’t look like part of this PR to me. Nothing’s changed here apart from dropping the window mock server side.